packages/webpack/src/plugin.ts

```diff
@@ -0,0 +1,224 @@
+        if (!request) {
+          return callback();
+        }
+
+        request.request = path.resolve(process.cwd(), 'app/ngfactory');
+        request.recursive =  true;
+        request.dependencies.forEach((d: any) => d.critical = false);
+        return callback(null, request);
+      });
+      cmf.plugin('after-resolve', (result: any, callback: (err?: any, request?: any) => void) => {
+        if (!result) {
+          return callback();
+        }
+
+        result.resource = this.options.genDir
+                     || path.resolve(process.cwd(), this.angularCompilerOptions.genDir + '/app');
+        result.recursive = true;
+        result.dependencies.forEach((d: any) => d.critical = false);
+        result.resolveDependencies = createResolveDependenciesFromContextMap((_: any, cb: any) => {
+          return cb(null, this.lazyRoutes);
+        });
+
+        return callback(null, result);
+      });
+    });
+
+    compiler.plugin('make', (compilation: any, cb: any) => this._make(compilation, cb));
+    compiler.plugin('after-emit', (compilation: any, cb: any) => {
+      this.done = null;
+      this.compilation = null;
+      compilation._ngToolsWebpackPluginInstance = null;
+      cb();
+    });
+  }
+
+  private _make(compilation: any, cb: (err?: any, request?: any) => void) {
+    const rootModulePath = path.normalize(this.rootModule + '.ts');
+    const rootModuleName = this.rootModuleName;
+    this.compilation = compilation;
+
+    if (this.compilation._ngToolsWebpackPluginInstance) {
+      cb(new Error('A ngtools/webpack plugin already exist for this compilation.'));
+    }
+    this.compilation._ngToolsWebpackPluginInstance = this;
+
+    this.loader = new WebpackResourceLoader(compilation);
+
+    const i18nOptions: any = {
+      i18nFile: undefined,
+      i18nFormat: undefined,
+      locale: undefined,
+      basePath: this.options.baseDir
+    };
+
+    // Create the Code Generator.
+    const codeGenerator = ngCompiler.CodeGenerator.create(
+      this.angularCompilerOptions,
+      i18nOptions,
+      this.program,
+      this.compilerHost,
+      new ngCompiler.NodeReflectorHostContext(this.compilerHost),
+      this.loader
+    );
+
+    // We need to temporarily patch the CodeGenerator until either it's patched or allows us
+    // to pass in our own ReflectorHost.
+    patchReflectorHost(codeGenerator);
+    this.done = codeGenerator.codegen()
+      .then(() => {
+        // process the lazy routes
+        const lazyModules = this._processNgModule(rootModulePath, rootModuleName, rootModulePath)
+          .map(moduleKey => moduleKey.split('#')[0]);
+        this.lazyRoutes = lazyModules.reduce((lazyRoutes: any, lazyModule: any) => {
+          lazyRoutes[`${lazyModule}.ngfactory`] = path.join(
+            path.resolve(process.cwd(), this.angularCompilerOptions.genDir),
+            'app', lazyModule + '.ngfactory.ts'
```

Comment:
i realized this is hardcoded and would fail for anything other than `app` - should be I guess the relative path between the ngfactory directory and the app's root.